report/4.Solution/LanuageOfCompiler.tex

```diff
@@ -0,0 +1,38 @@
+\section{The languages of the compiler}
+When creating a compiler there are three languages of relevance, as seen in the tombstone diagram\ref{fig:TStoneExample}.
+
+\begin{figure}[H]
+	\centering
+	\includegraphics[width=\textwidth/2+\textwidth/4]{4.Solution/images/T-diagram.png}
+	\caption{
+		Example of Tombstone diagram showcasing an example of three languages for a given compiler.\cite{TStoneWiki}
+	}
+	\label{fig:TStoneExample}
+\end{figure}
+In the case of the HCL compiler the three languages are as follows:
+
+\begin{enumerate}
+\item input: \textbf{HCL} \\
+\item output: \textbf{C++/C} \\
+\item compiler: \textbf{Kotlin} \\
+\end{enumerate}
+
+\subsection{Language of the compiler}
+The compiler was written in Kotlin\cite{KotlinWebsite}, a new programming language that compiles to both the JVM, javascript and LLVM. 
+The compiler is compiled to JVM because this grants access to all of the java ecosystem libraries, which in turn makes the JVM the most used target of kotlin.
+This choice to use kotlin was based on a number of factors.
+The group wanted to experiment with using a language they had no prior experience with for the duration of the project, but because of time limitations, a new language had to adhere to already known paradigms.
+The group knew C\# and a bit of Java, however as the HCL language has a lot of functional features, it was deemed important that the group became comfortable with a language that, at the very least was multi paradigm.
+Kotlin is inspired by both C\# and java and have a fairly similar syntax to that of those, which made it easier to learn from the perspective of the group.
+However it is still possible to write methods that is utilizes a functional domain.
+
+This grants the developer a large amount of possibilities, while still being relatively easy to learn.
+A disadvantage could be that low-level languages might have faster executing time, however as HCL is built with the Arduino architecture in mind, it is not possible to create vast programs, as the architecture has some hardware limitations, so execution time of the compiler will never really affect the end user, as the difference should not be noticeable.
+
+\subsection{Output language of the compiler}
+The group initially wanted to compile to assembly, primarily for educational purposes, however as the initial brainstorming took place, the group quickly realized that they wanted a big set of possibilities, which would waste a lot of development time on code generation. 
+In order to save time, the group agreed on compiling to C++, the language that Arduino software is normally written in.
+When writing code in the Arduino IDE, it gives access to a couple of predefined methods for interacting with I/O devices and then through a C/C++ compiler\cite{ArFAQ}.
```

Comment:
This sentence doesn't make sense after "and"
I'm not quite sure what was meant to be written here.